akka-stream-tests/src/test/scala/akka/stream/io/FileSourceSpec.scala

```diff
@@ -203,6 +225,19 @@ class FileSourceSpec extends StreamSpec(UnboundedMailboxConfig) {
       Await.result(r, 3.seconds.dilated).status.isFailure shouldBe true
     }
 
+    "onError with failure and return a failed IOResult when trying to read from a directory instead of a file" in assertAllStagesStopped {
+      val (r, p) = FileIO.fromPath(directoryInsteadOfFile).toMat(Sink.asPublisher(false))(Keep.both).run()
+      val c = TestSubscriber.manualProbe[ByteString]()
+      p.subscribe(c)
+
+      c.expectSubscription()
+
+      val error = c.expectError()
+      error shouldBe an[IllegalArgumentException]
+
+      Await.result(r, 3.seconds.dilated).status.isFailure shouldBe true
+    }
```

Comment:
Prefer `futureValue` where possible. It will pick up a reasonable default value. Awaits here are probably just a leftover before Scalatest had `futureValue`